File: test-annotate-helpers.js
/**
 * Helper module for annotate e2e test — functions without type annotations.
 * trickle annotate should add types to these functions after observation.
 */

function parseConfig(raw) {
  return {
    host: raw.host || 'localhost',
    port: raw.port || 3000,
    debug: raw.debug || false,
  };
}

function processItems(items) {
  return items.map(item => ({
    id: item.id,
    name: item.name.toUpperCase(),
    processed: true,
  }));
}

function calculateTotal(prices, taxRate) {
  const subtotal = prices.reduce((sum, p) => sum + p, 0);
  return { subtotal, tax: subtotal * taxRate, total: subtotal * (1 + taxRate) };
}

module.exports = { parseConfig, processItems, calculateTotal };

File: test_annotate_helpers.py
"""Helper module for annotate e2e test — Python functions without type annotations."""


def parse_config(raw):
    return {
        "host": raw.get("host", "localhost"),
        "port": raw.get("port", 3000),
        "debug": raw.get("debug", False),
    }


def process_items(items):
    return [{"id": item["id"], "name": item["name"].upper(), "processed": True} for item in items]


def calculate_total(prices, tax_rate):
    subtotal = sum(prices)
    return {"subtotal": subtotal, "tax": subtotal * tax_rate, "total": subtotal * (1 + tax_rate)}
